astropy/convolution/tests/test_convolve_fft.py

```diff
@@ -66,9 +68,11 @@ def test_unity_1_none(self, boundary, nan_treatment, normalize_kernel):
 
         y = np.array([1.], dtype='float64')
 
-        z = convolve_fft(x, y, boundary=boundary,
-                         nan_treatment=nan_treatment,
-                         normalize_kernel=normalize_kernel)
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore', AstropyUserWarning)
+            z = convolve_fft(x, y, boundary=boundary,
+                             nan_treatment=nan_treatment,
+                             normalize_kernel=normalize_kernel)
```

Comment:
But why is there a warning? I feel like if we catch them we should check that the warnings are ones we expect. What were the original warnings? Same comment in other places.

Reply:
That's why I tagged @keflavich to review. The actual warnings are available at https://travis-ci.org/astropy/astropy/jobs/443714275 . Or if that link no longer works by the time someone looks at it, you can remove `addopts = -p no:warnings` from `setup.cfg` and run the tests locally to see them.

There were a lot of `AstropyUserWarning: The convolve_fft version of boundary=None is equivalent to the convolve boundary='fill'.  There is no FFT equivalent to convolve's zero-if-kernel-leaves-boundary` and `RuntimeWarning: invalid value encountered in true_divide` but I have no idea if they are expected or not. These warnings, along with others, are currently silenced in CI.

Reply:
The former warning is supposed to be raised if `boundary=None` is given, since it is a point where convolve/convolve_fft are different.  The other one... is not desired but also not really important?  It probably indicates a div-by-zero error, which we don't care about because we handle NaNs/infs appropriately already.

Reply:
well, if it's expected to be raised then we should check whether it is, and if it's to be ignored as div by zero then we should just ignore it, rirght? Either way the tests are better to be cleaned up as we try not to have any warnings on CI to make our life easier to identify new issues.

Reply:
Agreed.  Probably the numpy warning catching should happen within the convolution code.

Reply:
Yeah there are indeed cases where we should actually just suppress the warning at the source.